data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/consumer/MultithreadedConsumer.java

```diff
@@ -0,0 +1,204 @@
+package org.opensearch.dataprepper.plugins.consumer;
+
+import java.time.Duration;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.consumer.CommitFailedException;
+import org.apache.kafka.clients.consumer.ConsumerRebalanceListener;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.errors.WakeupException;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.event.JacksonEvent;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.plugins.source.KafkaSourceConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MultithreadedConsumer implements Runnable, ConsumerRebalanceListener {
+
+	private static AtomicInteger recordCount = new AtomicInteger(0);
+	private final KafkaConsumer<String, String> consumer;
+	private static final Logger logger = LoggerFactory.getLogger(MultithreadedConsumer.class);
+	private final AtomicBoolean status = new AtomicBoolean(false);
+	private long lastCommitTime = System.currentTimeMillis();
+	private Map<TopicPartition, OffsetAndMetadata> currentOffsets = new HashMap<>();
+	private final KafkaSourceConfig sourceConfig;
+	@SuppressWarnings("deprecation")
+	private final Buffer<Record<Event>> buffer;
+	private long lastReadOffset = 0L;
+	private int count = 0;
+	private String consumerId;
+	private String consumerGroupId;
+	private Properties consumerProps;
+	private boolean firstRetryAttempt = false;
+
+	@SuppressWarnings("deprecation")
+	public MultithreadedConsumer(String consumerId, String consumerGroupId, Properties props,
+			KafkaSourceConfig sourceConfig, Buffer<Record<Event>> buffer) {
+		this.consumerProps = props;
+		this.consumer = new KafkaConsumer<>(consumerProps);
+		this.consumerId = consumerId;
+		this.consumerGroupId = consumerGroupId;
+		this.sourceConfig = sourceConfig;
+		this.buffer = buffer;
+
+	}
+
+	@Override
+	public void run() {
+		logger.info("Start reading ConsumerID:{}, Consumer Group: {}", consumerId, consumerGroupId);
+		try {
+			if (!sourceConfig.getTopicName().isEmpty() && isTopicAvailable(sourceConfig.getTopicName())) {
```

Comment:
This check should happen in the `KafkaSource`. That should allow the source to fail quickly rather than have all the threads throw errors.